crates/biome_configuration/src/editorconfig.rs

```diff
@@ -83,12 +83,14 @@ impl EditorConfig {
 #[derive(Debug, Clone, Deserialize, Default)]
 #[serde(default)]
 pub struct EditorConfigOptions {
-    indent_style: Option<IndentStyle>,
```

Comment:
Why not keep the same signature and directly call `T::default()` when we deserialize `unset`/`off`?

Reply:
Traceability. Imo, it adds minimal complexity here, and something like it is going to be necessary later. Might as well keep the information.

Since `serde_path_to_error` unfortunately doesn't work for us here, the parse failure diagnostics emitted don't indicate which field failed. That's for another PR though.

Reply:
Thanks! LGTM :)